src/RadixDlt.NetworkGateway.PostgresIntegration/Queries/EntityFungibleResourcesQuery.cs

```diff
@@ -0,0 +1,438 @@
+                    {
+                        var vaultNextCursorInclusive = f.Vaults.Values.SingleOrDefault(x => x.VaultNextCursorInclusive.HasValue)?.VaultNextCursorInclusive;
+                        string? vaultCursor = vaultNextCursorInclusive.ToGatewayModel()?.ToCursorString();
+                        var vaults = f
+                            .Vaults
+                            .Values
+                            .Where(x => !x.VaultFilterOut)
+                            .Select(
+                                v => new GatewayModel.FungibleResourcesCollectionItemVaultAggregatedVaultItem(
+                                    vaultAddress: v.VaultEntityAddress,
+                                    amount: v.VaultBalance.ToString(),
+                                    lastUpdatedAtStateVersion: v.VaultLastUpdatedAtStateVersion))
+                            .ToList();
+
+                        val = new GatewayModel.FungibleResourcesCollectionItemVaultAggregated(
+                            resourceAddress: f.ResourceEntityAddress,
+                            vaults: new GatewayModel.FungibleResourcesCollectionItemVaultAggregatedVault(
+                                totalCount: f.ResourceVaultTotalCount,
+                                nextCursor: vaultCursor,
+                                items: vaults),
+                            explicitMetadata: resourceExplicitMetadata);
+                    }
+                    else
+                    {
+                        val = new GatewayModel.FungibleResourcesCollectionItemGloballyAggregated(
+                            resourceAddress: f.ResourceEntityAddress,
+                            amount: f.ResourceBalance.ToString(),
+                            lastUpdatedAtStateVersion: f.ResourceLastUpdatedAtStateVersion,
+                            explicitMetadata: resourceExplicitMetadata);
+                    }
+
+                    return val;
+                })
+            .ToList();
+
+        return new GatewayModel.FungibleResourcesCollection(input.TotalFungibleResourceCount, fungibleCursor, fungibleItems);
+    }
+
+    private static async Task<IDictionary<long, PerEntityQueryResultRow>> ExecuteEntityFungibleResourcesQuery(
+        ReadOnlyDbContext dbContext,
+        IDapperWrapper dapperWrapper,
+        ICollection<long> entityIds,
+        long? resourceEntityId,
+        QueryConfiguration configuration,
+        CancellationToken token)
+    {
+        if (entityIds.Count == 0)
+        {
+            return ImmutableDictionary<long, PerEntityQueryResultRow>.Empty;
+        }
+
+        if (entityIds.Count > 1 && (resourceEntityId.HasValue || configuration.ResourceCursor != null || configuration.VaultCursor != null))
+        {
+            throw new InvalidOperationException("Neither resource filter nor cursors can be used if executing against multiple entities.");
+        }
+
+        var cd = new CommandDefinition(
```

Comment:
well that's how it was initially coded by Krzysztof, I think he spoke with you about that approach. I could potentially split that but it would result in extra round trip to database and since what we have right now seems to be performant i'd like to keep it as it is.

We already have 3 methods in that class one per endpoint 
- `Details`
- `FungibleResourcesPage`
- `FungibleResourceVaultsPage`

Same pattern is applied for NonFungible query.

Renamed that class to `EntityFungibleResourcesAndVaultsQuery`  and added comment on top of query method.

Reply:
Sounds good to me :)